libraries/AP_NavEKF/AP_SmallEKF.h

```diff
@@ -33,6 +33,8 @@
 
 #include <AP_Math/vectorN.h>
 
+#define MATH_CHECK_INDEXES 0
```

Comment:
AP_Math/vectorN.h depends on this being set to change the behavior of the header. And you are defining it *after* the header is included. So later if you try to define it to 1 nothing will change.